### PR TITLE
Service Collector Filter Flag

### PR DESCRIPTION
**Pull request by vbeausoleil**
Follow up to my comment in issue #105 

* Added flag to filter services (like the one for the process collector).
* Added example for said flag in README.md

### CODE REVIEW

**Comment by martinlindhe** [comment]
Looks perfect, thank you very much for your contribution!

File: collector/service.go [hunk 1]
```diff
@@ -14,15 +16,30 @@ func init() {
...
 // NewserviceCollector ...
 func NewserviceCollector() (Collector, error) {
 	const subsystem = "service"
+
+	var wc bytes.Buffer
+	if *serviceWhereClause != "" {
+		wc.WriteString("WHERE ")
+		wc.WriteString(*serviceWhereClause)
+	} else {
+		log.Println("warning: No where-clause specified for process collector. This will generate a very large number of metrics!")
```
Comment (carlpett) [inline collector/service.go]: I don't think we need this warning for the service collector, there won't be that many metrics.

**Comment by carlpett** [comment]
Bit late to the party here, sorry. @vbeausoleil would you be up for addressing the review point I added?

File: collector/service.go (hunk 1 above)
Comment (martinlindhe) [inline collector/service.go]: I think this was simply carried over from https://github.com/martinlindhe/wmi_exporter/blob/master/collector/process.go#L52

**Comment by vbeausoleil** [comment]
@carlpett Sure, I can address it if you wish. 

I felt like it was still a relevant warning since every service on a machine has 8 different states that are reported and exposed as metrics so it's still feels like a lot. Initially, that is the reason why I wanted to have a filter flag in order to reduce the noise/amount of metrics I scraped.

Just let me know if you want me to open another PR to remove the warning.

**Comment by carlpett** [comment]
Yes, on closer inspection you are right, there's a lot of them. So never mind then, and thanks for contributing! :)

**Comment by vbeausoleil** [comment]
Great! I'm currently looking into a way to enable service filters when the exporter is installed to make mass deployment simpler. I currently have a Powershell script that stops the service, changes the binary path to include the flags and starts it back up again.

I've been thinking about adding an install parameter.. Do you guys have any opinion on the subject?

**Comment by carlpett** [comment]
Sounds like a good approach. I think we might just want a blanket "extra flags" parameter to the installer rather than having a specialized one for service filters, disk filters, process filters, etc., though.

**Pull request merged**